src/main/java/hudson/plugins/sonar/SonarInstallation.java

```diff
@@ -76,24 +77,23 @@ public SonarInstallation(String name,
     String serverUrl, String serverAuthenticationToken,
     String mojoVersion, String additionalProperties, TriggersConfig triggers,
     String additionalAnalysisProperties) {
-    this(name, serverUrl, null, Secret.fromString(StringUtils.trimToNull(serverAuthenticationToken)),
+    this(name, serverUrl, null,
       mojoVersion, additionalProperties, additionalAnalysisProperties, triggers);
+    this.setServerAuthenticationToken(Secret.fromString(StringUtils.trimToNull(serverAuthenticationToken)));
   }
 
   @DataBoundConstructor
   public SonarInstallation(
     String name,
     String serverUrl,
     @Nullable String credentialsId,
-    @Nullable Secret serverAuthenticationToken,
```

Comment:
Breaks binary compatibility

Reply:
Afaik it should work OOTB if you switch to `@CheckForNull`

Reply:
I tried to revert my constructor change and change from `@Nullable` to `@CheckForNull`, but `CasCTest` failed.

```io.jenkins.plugins.casc.ConfiguratorException: serverAuthenticationToken is required to configure class hudson.plugins.sonar.SonarInstallation```

Reply:
https://github.com/jenkinsci/configuration-as-code-plugin/blob/f72058445c0250041303a73b5d8566140d337c67/plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java#L109-L137 BTW. It is actually an inverse way.

I;d guess the rejection is caused by the package-level anotation: https://github.com/SonarSource/sonar-scanner-jenkins/blob/22fd2881d4a8e247cc50fb79f80c67c2865f4d90/src/main/java/hudson/plugins/sonar/package-info.java which is enforced even when `CheckForNull` is set. I will create a ticket for JCasC

Reply:
https://github.com/jenkinsci/configuration-as-code-plugin/issues/1025 . Better to fix in JCasC IMHO

Reply:
If I comment out `@ParametersAreNonnullByDefault` from `package-info.java` the test passes with both annotations `@Nullable` and `@CheckForNull`. So I guess that actually should be resolved in CasC?

Reply:
Yup. Not a big deal for this particular issue

Reply:
I will close this PR then. Well, at least I learned about the correct test pattern ;-)